tests/test_w3d_math.cpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief Set of tests to validate w3d math components
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include <gtest/gtest.h>
+#include <vector2.h>
+#include <vector3.h>
+#include <vector4.h>
+
+#include <vector2i.h>
+#include <vector3i.h>
+
+#include <matrix3.h>
+#include <matrix4.h>
+
+TEST(w3d_math, vector2)
+{
+    Vector2 a(2.0f, 1.0f);
+
+    // Test length
+    EXPECT_FLOAT_EQ(a.Length(), GameMath::Sqrt(5.0f));
+    EXPECT_FLOAT_EQ(a.Length2(), 5.0f);
+
+    // Test normalize
+    auto tmp = a;
+    tmp.Normalize();
+    EXPECT_FLOAT_EQ(tmp.Length(), 1.0f);
+
+    Vector2 b(0.0f, 2.0f);
+
+    // Two notations for dot product
+    EXPECT_FLOAT_EQ(a.Dot_Product(a, a), 5.0f);
```

Comment:
`Dot_Product` is a static function no? I think it would be better to call as `Vector2::Dot_Product`.

Reply:
Right - fixed